### PR TITLE
Refactor slf4j logger 

### PR DESCRIPTION
**Pull request by basiliskus**
# Refactor slf4j logger 

- Refactored Slf4jLogger to avoid code duplication and for clarity
- Added Level enum to define logging levels
- Added FATAL logging level
- Changed code order to match logging level order
- Added tests for getLoggingEventBuilder helper method
- Added simplelogger.properties to configure the logger for the tests 

## Issue #84

### CODE REVIEW

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/external/slf4j/Slf4jLogger.java [hunk 1]
```diff
@@ -39,31 +57,42 @@ public void logInfo(String infoMessage, Object... parameters) {
...
-    public void logError(String errorMessage, Throwable e) {
-        LOGGER.atError().setMessage(() -> ANSI_RED + errorMessage + ANSI_RESET).setCause(e).log();
+    private LoggingEventBuilder getLoggingEventBuilder(Level level, String message) {
+        return switch (level) {
+            case TRACE -> LOGGER.atTrace().setMessage(() -> ANSI_PURPLE + message + ANSI_RESET);
+            case DEBUG -> LOGGER.atDebug().setMessage(() -> ANSI_CYAN + message + ANSI_RESET);
+            case INFO -> LOGGER.atInfo().setMessage(() -> ANSI_GREEN + message + ANSI_RESET);
+            case WARN -> LOGGER.atWarn().setMessage(() -> ANSI_YELLOW + message + ANSI_RESET);
+            case ERROR -> LOGGER.atError().setMessage(() -> ANSI_RED + message + ANSI_RESET);
+            case FATAL -> {
```
Comment (jorg3lopez) [inline app/src/main/java/gov/hhs/cdc/trustedintermediary/external/slf4j/Slf4jLogger.java:90]: I like the enum and method. What are your thoughts on adding a default case that will let us know it didn't match any of the cases?

**Review by jorg3lopez — APPROVED** [review]
Great job refactoring the logger logic.

File: app/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/slf4j/Slf4jLoggerTest.groovy [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger
+import gov.hhs.cdc.trustedintermediary.context.TestApplicationContext
+
+import org.slf4j.event.Level
+import org.slf4j.MarkerFactory
+
+import spock.lang.Specification
+
+
+class Slf4jLoggerTest extends Specification {
```
Comment (basiliskus) [inline app/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/slf4j/Slf4jLoggerTest.groovy:12]: I could not figure out how to set the log level for slf4j at runtime to test DEBUG and TRACE. Maybe if there is a way to create a `simplelogger.properties` files for the test folder like the one in `app/src/main/resources/simplelogger.properties`?

Comment (basiliskus) [reply app/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/slf4j/Slf4jLoggerTest.groovy:12]: Nevermind, I found out how to add the properties file for the tests and added the missing tests

**Review by jorg3lopez — APPROVED** [review]
Great Job!

**Pull request merged**